cmd/operator-sdk/run/packagemanifests/packagemanifests.go

```diff
@@ -59,3 +53,16 @@ must be set to a valid package manifests root directory, ex. '<project-root>/pac
 
 	return cmd
 }
+
+func (c *packagemanifestsCmd) validate(args []string) error {
+	if len(args) > 0 {
+		if len(args) > 1 {
+			return fmt.Errorf("exactly one argument is required")
+		}
+		c.ManifestsDir = args[0]
+	} else {
+		c.ManifestsDir = "packagemanifests"
+	}
```

Comment:
This should only be validating a `packagemanifestsCmd`, not setting fields
```suggestion
	if len(args) != 0 && len(args) > 1 {
		return fmt.Errorf("exactly one argument is required")
	}
```

Set defaults in a function like
```go
func (c *packagemanifestsCmd) setDefaults(args []string) {
	if len(args) != 0 {
		c.ManifestsDir = args[0]
	} else {
		c.ManifestsDir = "packagemanifests"
	}
}
```

Reply:
Yeah, I was trying to keep it simple by have all the pre-parsing stuff happen in validate(), I can break it out into a separate method.

Reply:
also, I might have to change it in some of the other parts, I think I stuck parsing into validate in some of the other refactors I did.